### ISSUE TITLE
[Feature Request]: Provide Setting to Use Types With Extra Result Wrapping

### ISSUE THREAD

**Issue by HappyHepo** [odata2ts/odata2ts#153]
As mentioned in #125, OData V2 services provided by SAP systems wrap expanded navigation properties in an extra object containing a results array.

While the workaround works for the generated client, the generated typings do not represent the SAP style.

Please add a setting for the generated typings instead of the service as suggested as initial workaround in #125.

**Comment by texttechne** [comment]
Actually not all V2 services (of SAP) are affected. As far as I've understood the issue, this is due to old ASP.net implementations...

To be crystal clear on your issue: You only need this for model generation, hence the types should reflect this extra result wrapping. Let's use the `Person` model of the TrippinService as example. Focussing only on the `trips` attribute, it would then look like this:
```ts
export interface PersonModel {
  trips?: { results: Array<TripModel> };
}

// in contrast to 
export interface PersonModel {
  trips?: Array<TripModel>;
}
```
Is this correct?

**Comment by HappyHepo** [comment]
This is correct.
The Netweaver Gateway and Java SDK services we are using all return data like the upper example.

**Comment by texttechne** [comment]
PR has been merged.

The configuration property is called `v2ModelsWithExtraResultsWrapping` and you need to set it to `true`.
This option only takes effect, when:
* `mode=models`
* the OData service in question is a V2 service

**Comment by texttechne** [comment]
Released with `@odata2ts/odata2ts` version `0.21.0`.

@HappyHepo Please test and close the issue if successful.

**Comment by HappyHepo** [comment]
Works for me
